Demo/MozillaSocial-iOS/MoSoContent/Auth.swift

```diff
@@ -0,0 +1,138 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import AuthenticationServices
+
+let isProd = false
+let baseURL = isProd ? "https://mozilla.social" : "https://stage.moztodon.nonprod.webservices.mozgcp.net"
+let redirectScheme = "mozillasocial"
+let redirectURI = redirectScheme + "://auth"
+
+class Auth: NSObject, ASWebAuthenticationPresentationContextProviding {
+
+    var authToken: Token?
+
+    func presentationAnchor(for session: ASWebAuthenticationSession) -> ASPresentationAnchor {
+        return ASPresentationAnchor()
+    }
+
+    func registerApp() async -> ClientEntity? {
+        var urlRequest = URLRequest(url: registerAppURL())
+        urlRequest.httpMethod = "POST"
+
+        do{
+            let (data, _) = try await URLSession.shared.data(for: urlRequest)
+
+            let decoder = JSONDecoder()
+            decoder.keyDecodingStrategy = .convertFromSnakeCase
+            return try decoder.decode(ClientEntity.self, from: data)
+        } catch {
+            print("error: ", error)
+            return nil
+        }
+    }
+
+    func signIn(client: ClientEntity) {
+        var authURL = signInURL(with: client.clientId)
+
+        let session = ASWebAuthenticationSession(url: authURL, callbackURLScheme: redirectScheme) { (url, error) in
+            guard error == nil else {
```

Comment:
because we are using `error` in the following statement, maybe we can unwrap it with an `if let` (instead of a `guard`) and skip the `String(describing: ...)`?